lib/elixir/lib/system.ex

```diff
@@ -471,6 +474,34 @@ defmodule System do
     :erlang.halt(String.to_charlist(status))
   end
 
+  @doc """
+  Carefully stops the Erlang runtime system.
+
+  All applications are taken down smoothly, all code is unloaded, and all ports
+  are closed before the system terminates by calling `halt/1`.
+
+  `status` must be a non-negative integer value which is returned by the
+  runtime system to the operating system.
```

Comment:
We say that the status should be a non-negative integer but we only check if it's an integer. I understand `:init.halt/1` will fail if the argument is a negative integer, but maybe we can collapse those checks here so that we'd see the `FunctionClauseError` for `System.stop(-1)`? Wdyt?